tests/integration/dns_test.py

```diff
@@ -204,6 +204,30 @@ def test_preserve_dns_config():
     assert dns_config == current_state[DNS.KEY][DNS.CONFIG]
 
 
+def test_preserve_dns_config_with_empty_state():
+    dns_config = {
+        DNS.SERVER: [IPV6_DNS_NAMESERVERS[0], IPV4_DNS_NAMESERVERS[0]],
+        DNS.SEARCH: []
+    }
+    desired_state = {
+        Interface.KEY: _get_test_iface_states(),
+        Route.KEY: {
+            Route.CONFIG: _gen_default_gateway_route()
+        },
+        DNS.KEY: {
+            DNS.CONFIG: dns_config
+        }
+    }
+    netapplier.apply(desired_state)
+    current_state = netinfo.show()
```

Comment:
Up until this point this fits a fixture, returning the last state.
Then the test should just apply the empty state and assert that the previous DNS config is still intact. (asserting on a new `show`, checking against the state from the fixture).